src/OpenLoco/src/Windows/TerraForm.cpp

```diff
@@ -1436,16 +1436,16 @@ namespace OpenLoco::Ui::Windows::Terraform
                 auto res = Ui::ViewportInteraction::getSurfaceLocFromUi({ x, y });
                 if (res)
                 {
-                    if (_adjustLandToolSize != 1)
+                    if (_adjustLandToolSize == 1 && !(isMountainMode || isPaintMode))
                     {
-                        auto count = TileManager::setMapSelectionTiles(res->first, 4);
+                        auto count = TileManager::setMapSelectionSingleTile(res->first, true);
 
                         if (!count)
```

Comment:
you can move these `if (count)` checks 1 level up outside this `if`

Reply:
I considered doing something similar, in that I thought the count could just be part of the `if` condition. Ultimately I decided against it, as refactoring wasn't the point of this PR.